src/app/Library/CrudPanel/Traits/Validation.php

```diff
@@ -397,4 +351,83 @@ private function getValidationRulesAndMessagesFromField($field, $parent = false)
 
         return [$rules, $messages];
     }
+
+    /**
+     * Return an array containing the request rules and the field/controller rules merged.
+     * The rules in request will take precedence over the ones in controller/fields.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  array  $rules
+     * @return array
+     */
+    private function mergeRules($request, $rules)
+    {
+        $extendedRules = [];
+        $requestRules = $this->getRequestRulesAsArray($request);
+        $rules = array_map(function ($ruleDefinition) {
+            return is_array($ruleDefinition) ? $ruleDefinition : explode('|', $ruleDefinition);
+        }, $rules);
+
+        foreach ($requestRules as $ruleKey => $rule) {
+            $extendedRules[$ruleKey] = array_key_exists($ruleKey, $rules) ? array_merge($rule, $rules[$ruleKey]) : $rule;
+            unset($rules[$ruleKey]);
+        }
+
+        return array_merge($rules, $extendedRules);
+    }
+
+    /**
+     * Return the request rules as an array of rules if developer provided a rule string configuration.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return array
+     */
+    private function getRequestRulesAsArray($request)
+    {
+        $requestRules = [];
+        foreach ($request->rules() as $ruleKey => $rule) {
+            $requestRules[$ruleKey] = is_array($rule) ? $rule : explode('|', $rule);
+        }
+
+        return $requestRules;
+    }
+
+    /**
+     * Checks if the request is valid against the rules.
+     *
+     * @param  array  $rules
+     * @param  array  $messages
+     * @param  \Illuminate\Http\Request|null  $request
+     * @return \Illuminate\Http\Request
+     */
+    private function checkRequestValidity($rules, $messages, $request = null)
```

Comment:
Note to self: this hasn't changed at all, just moved in a different place in the same file.